ovirt/provider.go

```diff
@@ -41,7 +41,6 @@ func Provider() terraform.ResourceProvider {
 			"headers": {
 				Type:        schema.TypeMap,
 				Optional:    true,
-				Default:     map[string]string{},
```

Comment:
@llomgui , any reason to remove this `Default` definition? IMHO, the program would go panic when it's not set by range over a `nil` value (https://github.com/imjoey/terraform-provider-ovirt/blob/master/ovirt/provider.go#L82).

Reply:
It creates an error: `panic: can't convert map[string]string{} to cty.Value`

We can set a Default: `DefaultFunc: schema.EnvDefaultFunc("OVIRT_HEADERS", nil),`
What do you think? @imjoey

Reply:
@llomgui The new `panic` seems like a compatible issue with `terraform 0.12.0`. Since the `header` is a `map`, a string representation for that stored in env may introduce more complexity for users. So your previous solution that remove the default is awesome. About the panic, we may need some more changes that will be discussed in others threads.

Reply:
@llomgui, I just committed a patch to your PR branch to improve the previous header processing, which could also eliminate the panic. Could you please help me review the code? Please let me know your insights on that, thank you.

Reply:
@imjoey LGTM